libs/pavex/src/response/response_.rs

```diff
@@ -110,6 +110,38 @@ impl Response {
         self
     }
 
+    /// Get a mutable reference to the [`Response`] status.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use pavex::http::StatusCode;
+    /// use pavex::response::Response;
+    /// use pavex::http::header::CONTENT_TYPE;
+    /// 
+    /// let mut response = Response::ok();
+    ///
+    /// // We're doing this trick with the brackets to tell Rust to drop the reference
+    /// // since you can't hold an immutable and mutable ref at the same time
+    /// {
+    ///     assert!(response.status() == StatusCode::OK);
+    /// }
+    /// {
+    /// // Get a mutable reference to the status.
+    /// let status = response.status_mut();
+    /// 
+    /// // Change the Status
+    /// *status = StatusCode::NOT_FOUND;
+    /// }
```

Comment:
I think this would compile without brackets, since the borrow checker is smart enough to figure out they don't overlap 🤔